ftw/usermigration/loginnames.py

```diff
@@ -0,0 +1,46 @@
+from Products.CMFCore.utils import getToolByName
+from Products.PluggableAuthService.interfaces.plugins import IUserEnumerationPlugin
+from Products.PluggableAuthService.plugins.ZODBUserManager import IZODBUserManager
+
+
+def migrate_loginnames(context, mapping, mode='move', replace=False):
+    """Migrate Plone loginnames.
+    """
+
+    # Statistics
+    moved = []
+    copied = []
+    deleted = []
+
+    uf = getToolByName(context, 'acl_users')
+
+    for old_userid, new_userid in mapping.items():
+        for plugin_id, plugin in uf.plugins.listPlugins(IUserEnumerationPlugin):
+            # Only ZODB User Manager is supported
+            if not IZODBUserManager.providedBy(plugin):
+                continue
+
+            for user in plugin.enumerateUsers(id=old_userid, exact_match=True):
```

Comment:
The login names migration is a little bit special. But I would expect that the provided mapping contains an 'old login' -> 'new login' mapping and not a 'current userid' -> 'new login' one. If login and userid are the same, it doesn't matter. If they are different, then the logins migration has to be run separately in any case.